Source/WTF/wtf/Compiler.h

```diff
@@ -456,14 +456,15 @@
 #if COMPILER(CLANG)
 #define IGNORE_CLANG_WARNINGS_BEGIN(warning) IGNORE_WARNINGS_BEGIN_IMPL(clang, warning)
 #define IGNORE_CLANG_WARNINGS_END IGNORE_WARNINGS_END_IMPL(clang)
+#define IGNORE_WARNINGS_BEGIN(warning) IGNORE_CLANG_WARNINGS_BEGIN(warning)
+#define IGNORE_WARNINGS_END IGNORE_CLANG_WARNINGS_END
 #else
 #define IGNORE_CLANG_WARNINGS_BEGIN(warning)
 #define IGNORE_CLANG_WARNINGS_END
+#define IGNORE_WARNINGS_BEGIN(warning) IGNORE_GCC_WARNINGS_BEGIN(warning)
+#define IGNORE_WARNINGS_END IGNORE_GCC_WARNINGS_END
 #endif
 
-#define IGNORE_WARNINGS_BEGIN(warning) IGNORE_WARNINGS_BEGIN_IMPL(GCC, warning)
-#define IGNORE_WARNINGS_END IGNORE_WARNINGS_END_IMPL(GCC)
```

Comment:
What's the goal of this? Now we no longer benefit from suppressing -Wpragmas.

Reply:
`IGNORE_WARNINGS_BEGIN` and its counterpart would expand to either the GCC or clang impl, depending on the selected compiler.

Before this change it would expand to the GCC impl and be useless when clang is the selected compiler.

Reply:
No: Clang respects the GCC pragmas and they are emitted unconditionally. I don't think the change is doing anything.

(I did misread the code, though. I see -Wpragmas is still suppressed, so this isn't hurting anything either.)

Reply:
> Clang respects the GCC pragmas and they are emitted unconditionally. 

Why do you think I am sending this PR?
With the current main, clang and ASan enabled the warnings DO show up.

Reply:
They are showing up because the current code uses `IGNORE_GCC_WARNINGS_BEGIN`/`END` which indicates a desire to not ignore the warning if building with Clang.

All you need to do is change that to `IGNORE_WARNINGS_BEGIN`/`END` and then it will work as you desire.

Reply:
So what I'm saying is your changes in Vector.h should be sufficient.

`pragma GCC` is respected by both GCC and Clang.

`pragma clang` is respected by Clang only.

Switching to `pragma clang` when building with Clang, while harmless, also doesn't actually solve anything, since if building with Clang they're both equivalent.

Reply:
Understood, thanks!